querybuilder/rule_group.go

```diff
@@ -6,36 +6,40 @@ const (
 )
 
 type Checker interface {
-	Evaluate(dataset map[string]interface{}) bool
+	Evaluate(dataset map[string]interface{}) (bool, error)
 }
 
 type RuleGroup struct {
 	Condition interface{}
 	Rules     interface{}
 }
 
-func (rg *RuleGroup) Evaluate(dataset map[string]interface{}) bool {
+func (rg *RuleGroup) Evaluate(dataset map[string]interface{}) (bool, error) {
 	rules := rg.Rules.([]interface{})
-	var res bool
+
 	switch rg.Condition.(string) {
 	case AND:
 		for _, r := range rules {
-			res = rg.getChecker(r.(map[string]interface{})).Evaluate(dataset)
-			if !res {
-				return false
+			getCheckerResponse := rg.getChecker(r.(map[string]interface{}))
+			evaluationResponse, err := getCheckerResponse.Evaluate(dataset)
+			if err != nil || !evaluationResponse {
+				return evaluationResponse, err
 			}
 		}
-		return true
+		return true, nil
+
 	case OR:
 		for _, r := range rules {
-			res = rg.getChecker(r.(map[string]interface{})).Evaluate(dataset)
-			if res {
-				return true
+			getCheckerResponse := rg.getChecker(r.(map[string]interface{}))
+			evaluationResponse, err := getCheckerResponse.Evaluate(dataset)
+			if err != nil || evaluationResponse {
+				return evaluationResponse, err
 			}
 		}
-		return false
+		return false, nil
```

Comment:
In go, variable names are usually short and descriptive
so `getCheckerResponse` and `evaluationResponse` look out of place here. They're not even doing any network call so the term response doesn't make sense.